addons/website_event_sale/controllers/main.py

```diff
@@ -79,3 +89,12 @@ def registration_confirm(self, event, **post):
                 request.website.sale_reset()
 
         return res
+
+    def registration_filter(self, registration_data):
+        ticket_ids = {data['event_ticket_id'] for data in registration_data if data.get('event_ticket_id')}
+        registration_tickets = request.env['event.event.ticket'].sudo().browse(ticket_ids)
```

Comment:
Probably better to do a domain search with IDs "in" ticket_ids, price = 0.

Then check if the ticket is inside the result.